giraffe/src/components/geo/processing/NativeGeoTable.ts

```diff
@@ -0,0 +1,89 @@
+// Types
+import {AbstractGeoTable, CoordinateEncoding} from './AbstractGeoTable'
+import {Table} from '../../../types'
+import {Track} from './GeoTable'
+
+// Constants
+import {
+  GEO_HASH_COLUMN,
+  LAT_COLUMN,
+  LON_COLUMN,
+  TABLE_COLUMN,
+} from './tableProcessing'
+
+export class NativeGeoTable extends AbstractGeoTable {
+  table: Table
+  maxRows: number
+
+  constructor(table: Table, maxRows: number) {
+    super(getDataEncoding(table))
+    this.table = table
+    this.maxRows = maxRows
+  }
+
+  getRowCount() {
+    return Math.min(this.table.length, this.maxRows)
+  }
+
+  getValue(index: number, field: string): number {
+    const column = this.table.getColumn(field)
+    if (!column) return null
```

Comment:
Please always use braces for if statements. We will be updating the eslint rules to reflect this.